### PR TITLE
Add clean command to fireci

### PR DESCRIPTION
**Pull request by daymxn**
Per [b/381270432](https://b.corp.google.com/issues/381270432),

This adds a command to our `fireci` cli tool that can work as an alternative to `gradle clean`. There seems to be some dependency issues preventing `gradle clean` from completely correctly, but with the usage of `fireci`- we can avoid working within gradle entirely.

Furthermore, this command comes with flags for deeper cleaning of gradle's caches.

I've also updated the dependencies that fireci was using, as I ran into a variety of issues trying to use it "as-is" on an M1 machine. The updated dependencies seem to include fixes for such issues.

Command usage:
```sh
fireci clean --help
```

### CODE REVIEW

File: ci/fireci/fireci/ci_utils.py [hunk 1]
```diff
@@ -61,3 +63,9 @@ def gcloud_identity_token():
   """Returns an identity token with the current gcloud service account."""
   result = subprocess.run(['gcloud', 'auth', 'print-identity-token'], stdout=subprocess.PIPE, check=True)
   return result.stdout.decode('utf-8').strip()
+
+def get_projects(file_path: str = "subprojects.cfg") -> List[str]:
+  """Parses the specified file for a list of projects in the repo."""
+  with open(file_path, 'r') as file:
```
Comment (rlazo) [inline ci/fireci/fireci/ci_utils.py:69]: You can use the config parser module for this

```
import configparser

config = configparser.ConfigParser(delimiters=('='), allow_unnamed_section=True, allow_no_value=True)
config.read('subprojects.cfg')
return [k for k,v in config.items(configparser.UNNAMED_SECTION)]
```

File: ci/fireci/fireci/dir_utils.py [hunk 2]
```diff
@@ -30,3 +33,28 @@ def chdir(directory):
...
+    _logger.debug(f"Directory already deleted: {dir}")
+    return False
+
+  _logger.debug(f"Deleting directory: {dir}")
+  shutil.rmtree(dir)
+  return True
+
+def rmglob(pattern: str) -> int:
+  """Deletes all files that match a given pattern, and returns the amount of (root) files deleted"""
+  files = 0
```
Comment (rlazo) [inline ci/fireci/fireci/dir_utils.py:50]: `files` will be the size of `glob.glob`, no need for a counter

File: ci/fireci/fireci/dir_utils.py [hunk 3]
```diff
@@ -30,3 +33,28 @@ def chdir(directory):
...
+  for file in glob.glob(os.path.expanduser(pattern)):
+    path = pathlib.Path(file)
+    if path.is_dir():
+      rmdir(file)
+    else:
+      _logger.debug(f"Deleting file: {path}")
+      os.remove(path)
+    files += 1
+
+  return files
```
Comment (rlazo) [inline ci/fireci/fireci/dir_utils.py:60]: Out of curiosity, why you need the count?

File: ci/fireci/fireciplugins/clean.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+
+  cache = cache or deep
+  gradle = gradle or cache
+
+  cleaners = []
+
+  if build:
+    cleaners.append(delete_build)
+  if gradle:
+    cleaners.append(delete_build)
```
Comment (rlazo) [inline ci/fireci/fireciplugins/clean.py:66]: both conditions run the same code

File: ci/fireci/fireci/dir_utils.py (hunk 2 above)
Comment (daymxn) [reply ci/fireci/fireci/dir_utils.py:50]: fixed

File: ci/fireci/fireciplugins/clean.py (hunk 4 above)
Comment (daymxn) [reply ci/fireci/fireciplugins/clean.py:66]: ope, fixed. good catch!

File: ci/fireci/fireci/ci_utils.py (hunk 1 above)
Comment (rlazo) [reply ci/fireci/fireci/ci_utils.py:69]: Ouch, 3.9 is old, we should bump that... but not all the way to 3.13.

Feel free to ignore the suggestion.

File: ci/fireci/fireci/dir_utils.py (hunk 3 above)
Comment (daymxn) [reply ci/fireci/fireci/dir_utils.py:60]: To display at the end of the `clean` run. It makes it easier to see the result if you have `debug` logging disabled.

example

**Review by rlazo — APPROVED** [review]

**Pull request merged**